### PR TITLE
[dev] Handles cf-ew-raw- headers

### LINKED ISSUE #873: [dev] add cf-ew headers to request
The preview service will send any headers prefixed with `cf-ew-` along to the worker so we should make the proxy modify these headers before sending them.

### PR DESCRIPTION
**Pull request by EverlastingBugstopper**
Fixes #873 

worker code returns stringified headers in the response body:

```js
async function handleRequest(request) {
  let headers = new Map(request.headers)
  return new Response(mapToString(headers))
}
```

```console
$ curl http://localhost:8000 -H "Hi: There"
{"accept":"*/*","hi":"There","host":"example.com","user-agent":"curl/7.64.1","x-forwarded-proto":"https"}
```

### CODE REVIEW

File: src/commands/dev/mod.rs [hunk 1]
```diff
@@ -88,6 +88,17 @@ fn preview_request(
     let now: DateTime<Local> = Local::now();
     let preview_id = &preview_id;
 
+    let mut headers: HeaderMap = HeaderMap::new();
+
+    for header in &parts.headers {
+        let (name, value) = header;
+        let forward_header = format!("cf-ew-raw-{}", name);
+        // TODO: remove unwrap
+        let header_name = HeaderName::from_bytes(forward_header.as_bytes()).unwrap();
```
Comment (gabbifish) [inline src/commands/dev/mod.rs]: This function returns a Result with error type failure::Error, can we just get rid of this unwrap now? If the error type returned by calling `?` does not work with failure::Error, we can just capture the error and wrap it in a failure::Error

Comment (EverlastingBugstopper) [reply src/commands/dev/mod.rs]: This function actually returns a ResponseFuture, and I'm not quite sure how to make that play nicely with the failure crate which is why i have the TODO statements. cc @ashleygwilliams

File: src/commands/dev/mod.rs [hunk 2]
```diff
@@ -47,12 +49,19 @@ pub async fn dev(
...
             Ok::<_, failure::Error>(service_fn(move |req| {
-                preview_request(
-                    req,
-                    client.to_owned(),
-                    preview_id.to_owned(),
-                    server_config.clone(),
-                )
+                let client = client.to_owned();
+                let preview_id = preview_id.to_owned();
+                let server_config = server_config.clone();
```
Comment (gabbifish) [inline src/commands/dev/mod.rs]: do we need to clone this twice? (once in line 49, once here?) I feel like we can put variables like client, preview_id, and server_config in the arguments section `move |args|` and avoid unnecessary clones :)

File: src/commands/dev/mod.rs [hunk 3]
```diff
@@ -65,6 +74,22 @@ pub async fn dev(
     Ok(())
 }
 
+fn munge_response_headers(parts: &mut Parts) -> Result<(), failure::Error> {
```
Comment (gabbifish) [inline src/commands/dev/mod.rs]: nit: could we use a term more specific than munge, like `strip_cf_prefix_from_headers`?

File: src/commands/dev/mod.rs [hunk 4]
```diff
@@ -89,7 +114,16 @@ fn preview_request(
     let now: DateTime<Local> = Local::now();
     let preview_id = &preview_id;
 
-    parts.uri = get_preview_url(&path).expect("Could not get preview url");
+    let mut headers: HeaderMap = HeaderMap::new();
```
Comment (gabbifish) [inline src/commands/dev/mod.rs]: maybe factor this header prefix addition logic out into another function that acts as an "opposite" function to `munge_response_headers`?

**Pull request merged**